src/annotator/hypothesis-injector.js

```diff
@@ -0,0 +1,73 @@
+import FrameObserver from './frame-observer';
+import * as frameUtil from './util/frame-util';
+
+/**
+ * @typedef {import('../shared/bridge').default} Bridge
+ * @typedef {import('../types/annotator').Destroyable} Destroyable
+ */
+
+/**
+ * HypothesisInjector has logic for injecting Hypothesis client into iframes that
+ * are added to the page if (1) they have the `enable-annotation` attribute set
+ * and (2) are same-origin with the current document.
+ *
+ * @implements Destroyable
+ */
+export class HypothesisInjector {
+  /**
+   * @param {Element} element
+   * @param {Bridge} bridge - Channel for communicating with the sidebar
+   * @param {Record<string, any>} config - Annotator configuration that is
+   *   injected, alone with the Hypothesis client, into the the children iframe/s
+   */
+  constructor(element, bridge, config) {
+    this._bridge = bridge;
+    this._config = config;
+    /** @type {Map<HTMLIFrameElement, string>} */
+    this._frameIdentifiers = new Map();
+    this._frameObserver = new FrameObserver(
+      element,
+      frame => this._injectIntoFrame(frame),
+      frame => this._iframeUnloaded(frame)
+    );
+  }
+
+  /**
+   * Disables the injection of the Hypothesis client into children iframe/s.
```

Comment:
This is now a useful comment 👍 

```suggestion
   * Disables the injection of the Hypothesis client into child iframes.
```